src/clj/fluree/db/query/exec/select.cljc

```diff
@@ -15,30 +15,65 @@
 
 #?(:clj (set! *warn-on-reflection* true))
 
+(defn var-name
+  "Stringify and remove q-mark prefix of var for SPARQL JSON formatting."
+  [var]
+  (subs (name var) 1))
+
+(defn disaggregate
+  "For SPARQL JSON results, no nesting of data is permitted - the results must be
+  tabular. This function unpacks a single result into potentially multiple 'rows' of
+  results."
+  [result]
+  (let [aggregated (filter (fn [[k v]] (sequential? v)) result)]
+    (loop [[[agg-var agg-vals] & r] aggregated
+           results [result]]
+      (if agg-var
+        (let [results* (reduce (fn [results* result]
+                                 (into results* (map (fn [v] (assoc result agg-var v)) agg-vals)))
+                               []
+                               results)]
+          (recur r results*))
+        results))))
+
 (defmulti display
   "Format a where-pattern match for presentation based on the match's datatype.
   Return an async channel that will eventually contain the formatted match."
-  (fn [match _compact]
+  (fn [match output-format _compact]
```

Comment:
This new `output-format` binding exists only to provide runtime polymorphism. This implementation also litters this `output-format` argument throughout the call stack forcing already working existing code to be modified to support this new feature.

 We should use the built in language facilities to support run time polymorphism instead of rolling our own. Either new implementations of existing protocols/multi-methods, or a new protocol or multimethod would be a better fit.

Reply:
My current plan is to change the `where/match` into a record and implement a display method for it. I've got it work-in-progress for `construct` support, but didn't want to block Nexus and CAM work while I was still figuring out the proper abstraction.